sw/host/tests/rom/e2e_bootstrap_rma/src/main.rs

```diff
@@ -18,362 +21,315 @@ use opentitanlib::dif::lc_ctrl::{
-for { set i 0 } { $i < 1000 } { set i [expr {$i + 1}] } {
-    set reg_value [ lc_ctrl.tap.0 riscv dmi_read 0x6 ]
-    if { $reg_value == 4660 } {
-        break
+    match result {
+        ExitStatus::ExitSuccess => Ok(()),
+        result => bail!("FAIL: {result:?}"),
     }
-    sleep 5
 }
-if { $i == 1000 } {
-    echo "Ran out of attempts before TransitionToken0 became 0x1234"
-    nonexistent_command_that_causes_openocd_to_exit_with_error
-}"#
-        );
-        assert_eq!(
-            OpenocdTclBlock::WriteReg(LcCtrlReg::TransitionToken3, 0xabcd).into_tcl(),
-            "lc_ctrl.tap.0 riscv dmi_write 0x9 0xabcd"
-        );
-
-        assert_eq!(OpenocdTclBlock::Echo("bar").into_tcl(), "echo \"bar\"");
+
+/// Reset the chip with a specified strapping configuration.
+///
+/// The given `strapping_confs` are names of known strapping configurations.
+/// See [`TransportWrapper::pin_strapping`] for details.
+fn reset_with_strapping(
+    transport: &TransportWrapper,
+    strapping_confs: &[&str],
+    delay: Duration,
+) -> anyhow::Result<()> {
+    log::info!("Resetting target...");
+
+    // Apply the given strappings.
+    for strapping in strapping_confs {
+        transport
+            .pin_strapping(strapping)
+            .context(format!("failed to find strapping {strapping}"))?
+            .apply()
+            .context(format!("failed to apply strapping {strapping}"))?;
     }
 
-    /// Stringifying echo statements should panic when they contain unsupported
-    /// characters.
-    #[test]
-    #[should_panic]
-    fn openocd_statements_echo_panic() {
-        let _s = OpenocdTclBlock::Echo("bad string \"").into_tcl();
+    // Reset to see the affects.
+    transport
+        .reset_target(delay, true)
+        .context("failed to reset target")?;
+
+    // Remove the strappings so future resets are "clean".
+    for strapping in strapping_confs {
+        transport
+            .pin_strapping(strapping)
+            .context(format!("failed to find strapping {strapping}"))?
+            .remove()
+            .context(format!("failed to remove strapping {strapping}"))?;
     }
+
+    Ok(())
+}
+
+// FIXME: This is useful, move to `test_lib`.
+/// Wait for the lifecycle controller to enter the `READY` state.
+///
+/// Will return an error if the lifecycle controller enters an error stats, or
+/// does not enter `READY` within the given timeout.
+fn wait_until_ready(jtag: &Rc<dyn Jtag>, timeout: Duration) -> anyhow::Result<()> {
+    const POLL_DELAY: Duration = Duration::from_millis(5);
+
+    poll::poll_until(timeout, POLL_DELAY, || {
+        let status = jtag
+            .read_lc_ctrl_reg(&LcCtrlReg::Status)
+            .context("failed to read STATUS register")?;
+
+        if status & LcCtrlStatusBit::errors() != 0 {
+            bail!("lifecycle controller has error status: {status:#b}");
+        }
+
+        let ready_mask = LcCtrlStatusBit::Ready.mask();
+        Ok(status & ready_mask == ready_mask)
+    })
 }
```

Comment:
TODO: Move this somewhere useful, and probably update it to accept a status mask to check for rather than just `READY`.